### PR TITLE
config, api: Change audit switch into PDServerConfig

### LINKED ISSUE #4761: Persistent Audit switch config
## Enhancement Task
As title.

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: Close #4761
Persistent audit switch

### What is changed and how it works?
Change audit switch into PDServerConfig and add json/toml tag.

```commit-message
Change audit switch into PDServerConfig to persistent it
```

### Check List

Tests

- Unit test
- Integration test

Code changes

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/server/api/api_test.go [hunk 1]
```diff
@@ -146,15 +146,20 @@ func (s *testMiddlewareSuite) TestRequestInfoMiddleware(c *C) {
 	c.Assert(failpoint.Enable("github.com/tikv/pd/server/api/addRequestInfoMiddleware", "return(true)"), IsNil)
 	leader := s.cluster.GetServer(s.cluster.GetLeader())
 
-	req, _ := http.NewRequest("POST", leader.GetAddr()+"/pd/api/v1/admin/audit-middleware?enable=true", nil)
```
[the remark this answers is not in the document]
Comment (CabinfeverB) [reply tests/server/api/api_test.go:149]: IMO, we can remove it.

File: server/api/admin.go [hunk 2]
```diff
@@ -153,6 +153,6 @@ func (h *adminHandler) SwitchAuditMiddleware(w http.ResponseWriter, r *http.Requ
 		h.rd.JSON(w, http.StatusBadRequest, "The input is invalid.")
 		return
 	}
-	h.svr.SetAuditMiddleware(enable)
+	updatePDServerConfig(h.svr, h.svr.GetConfig(), "enable-audit", &enable)
```
Comment (AndreMouche) [inline server/api/admin.go:156]: First of all, should we consider failed case caused by `h.svr.SetPDServerConfig(..)` (for example persist failed)?
And I think with using` json.Marshal` and `json.Umarshal` makes the update too conflicted. how about updating and persisting the config directly(which should be more friendly for the maintainers)?

```suggestion
	    cfg := h.svr.GetConfig().PDServerCfg
	if cfg.EnableAudit == enable {
		// TODO:
		return
	}
	cfg.EnableAudit = enable
	err = h.svr.SetPDServerConfig(cfg)
	if err != nil {
		//TODO:
		return
	}
	h.rd.JSON(w, http.StatusOK, "Switching audit middleware is successful.")
```

File: tests/server/api/api_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (CabinfeverB) [reply tests/server/api/api_test.go:149]: > Should we keep `/admin/audit-middleware`?

I have remove it

File: server/api/admin.go (hunk 2 above)
Comment (CabinfeverB) [reply server/api/admin.go]: I have deleted this api

File: server/api/config.go [hunk 3]
```diff
@@ -154,7 +154,7 @@ func (h *confHandler) updateConfig(cfg *config.Config, key string, value interfa
 		}
 		return h.updateReplicationModeConfig(cfg, kp[1:], value)
 	case "pd-server":
-		return h.updatePDServerConfig(cfg, kp[len(kp)-1], value)
+		return updatePDServerConfig(h.svr, cfg, kp[len(kp)-1], value)
```
Comment (rleungx) [inline server/api/config.go:157]: Why change it?

File: tests/server/api/api_test.go [hunk 4]
```diff
@@ -471,3 +530,17 @@ func mustRequestSuccess(c *C, s *server.Server) http.Header {
 	c.Assert(resp.StatusCode, Equals, http.StatusOK)
 	return resp.Header
 }
+
+func mustWaitLeader(c *C, svrs map[string]*tests.TestServer) *server.Server {
```
Comment (rleungx) [inline tests/server/api/api_test.go:534]: We have many `mustWaitLeader` inside our repo, maybe we can reduce them.

**Review by rleungx — APPROVED** [review]
The rest LGTM

File: server/api/config.go (hunk 3 above)
Comment (CabinfeverB) [reply server/api/config.go:157]: Because I want to use func `updatePDServerConfig` in `admin.go` before.

**Review by nolouch — COMMENTED** [review]
lgtm.

**Review by nolouch — APPROVED** [review]

File: server/api/config.go (hunk 3 above)
Comment (CabinfeverB) [reply server/api/config.go]: reset it

**Pull request merged**